### ISSUE TITLE
Move the bounding box to the ref file for MIRI Imager

### ISSUE THREAD

**Issue by nden** [spacetelescope/jwst#1604]
The bounding box for the MIRI imager should be attached to the distortion transform in the reference file.
The assign_wcs step will change the bounding box for subarrays but otherwise will use the one that comes with the distortion transform. This can be done as part of the next reference file delivery to CRDS.

**Comment by jdavies-st** [comment]
It seems this work is done.

Using the current latest MIRI distortion reffile from CRDS:
```
In [9]: a = datamodels.DistortionModel('jwst_miri_distortion_0023.asdf')

In [10]: a
Out[10]: <DistortionModel from jwst_miri_distortion_0023.asdf>

In [11]: a.model
Out[11]: <CompoundModel137(offset_0=-4., c0_2=-12.78750038, c1_2=0.025, c0_3=-12.78750038, c1_3=0.025, c0_0_5=0.00000647, c1_0_5=-0.00020063, c2_0_5=0.00002433, c3_0_5=0.00000219, c4_0_5=-0.00000025, c0_1_5=2.82207821, c0_2_5=-0.00002873, c0_3_5=0.00000244, c0_4_5=0.00000016, c1_1_5=-0.00167748, c1_2_5=0.00000198, c1_3_5=0.00000181, c2_1_5=-0.00003313, c2_2_5=-0.00000019, c3_1_5=0.00000226, c0_0_6=0.00027793, c1_0_6=2.84373271, c2_0_6=0.00011154, c3_0_6=-0.00011824, c4_0_6=0.00000331, c0_1_6=0.00001385, c0_2_6=-0.00083572, c0_3_6=-0.00000011, c0_4_6=0.00000049, c1_1_6=-0.00001816, c1_2_6=-0.00006579, c1_3_6=0.00000018, c2_1_6=0.00000107, c2_2_6=0.0000033, c3_1_6=-0.00000026, c0_0_8=-1.54000068, c1_0_8=-0.02608358, c0_1_8=-0.00202991, c0_0_9=-7.22800207, c1_0_9=0.00202991, c0_1_9=-0.02608359, factor_13=-1., offset_14=-7.8, factor_15=60., factor_16=60.)>

In [12]: a.model.bounding_box
Out[12]: ((-0.5, 1023.5), (3.5, 1027.5))
```

**Comment by jdavies-st** [comment]
Also tracked by [JP-232](https://jira.stsci.edu/browse/JP-232).

The question has come up as to wether the resampled output should resample the whole detector or just the imaging area.  Currently, it keys off the bounding box in the distortion reference file, which is set to the full detector.

If it is decided that only the imaging area should be resampled, then the bounding box in the distortion file should be updated.

**Comment by nden** [comment]
This was fixed in #2980
